### PR TITLE
fix: Failing deploy of lambdas which attach to Event Bridge against "default" stage fails

### PR DESCRIPTION
**Pull request by TheEdgeOfCat**
Closes: #7994

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/eventBridge/index.js [hunk 1]
```diff
@@ -147,6 +147,10 @@ class AwsCompileEventBridgeEvents {
                     ],
                   ],
                 };
+                let fullRuleName = `rule/${eventBusName}/${RuleName}`;
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/eventBridge/index.js:150]: Let's maybe define it with `const` as  `(eventBusName === 'default') ? .. : ...`

**Comment by medikoo** [comment]
Additionally it'll be great to add regression test, best if written with `runServerless`. See https://github.com/serverless/serverless/tree/master/tests#unit-tests

**Comment by TheEdgeOfCat** [comment]
I have added the test case to the integration test. I am not sure whether unit tests are also necessary, but let me know if I should add that as well.

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js [hunk 2]
```diff
@@ -750,6 +750,119 @@ describe('AwsCompileEventBridgeEvents', () => {
...
+                    eventSource: ['cloudformation.amazonaws.com'],
+                  },
+                },
+              },
+            },
+          ],
+        },
+      };
+
+      return expect(awsCompileEventBridgeEvents.compileEventBridgeEvents()).to.be.fulfilled.then(
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/eventBridge/index.test.js:775]: All new unit tests should be written with `runServerless`. Check: https://github.com/serverless/serverless/tree/master/tests#unit-tests, you may use extend feature to extend basic `function` fixture and tests against that.

e.g. it's how tests were configured for latest ALB addition: https://github.com/serverless/serverless/blob/04d463f4e438d02fef958fce34677a06645fee82/lib/plugins/aws/package/compile/events/alb/lib/healthCheck.test.js

File: tests/integration-all/event-bridge/service/serverless.yml [hunk 3]
```diff
@@ -13,6 +13,14 @@ functions:
           pattern:
             source:
               - serverless.test
+  eventBusDefaultArn:
```
Comment (medikoo) [inline tests/integration-all/event-bridge/service/serverless.yml:16]: Great move, on covering that with integration tests!

Comment (TheEdgeOfCat) [reply tests/integration-all/event-bridge/service/serverless.yml:16]: I tested the integration tests on my AWS account and it created everything just fine. I have also run the deployment on our app with my modified code and that worked as well.

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js (hunk 2 above)
Comment (TheEdgeOfCat) [reply lib/plugins/aws/package/compile/events/eventBridge/index.test.js]: I think I got it now, though I didn't have time to convert all the existing test cases to the runServerless method. I'll try to get around to it by tomorrow. Should I make a new pull request for that?

Comment (TheEdgeOfCat) [reply lib/plugins/aws/package/compile/events/eventBridge/index.test.js]: Hmm, seems like the test fails on windows. There is no error message or anything to indicate what could have gone wrong :frowning_face: 

I don't have a windows machine to check this

**Comment by TheEdgeOfCat** [comment]
Also @medikoo I'm sorry for bothering you with this. I really appreciate your help and I'll try to make it up by helping around the project when I can.

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js (hunk 2 above)
Comment (medikoo) [reply lib/plugins/aws/package/compile/events/eventBridge/index.test.js]: > Hmm, seems like the test fails on windows

It's very weird error. I've run your branch on my Windows instance and tests have run successfully. To be honest I have not clear idea what happens there (especially that there's no error log, for unknown reason).

However I've patched one issue which prevented an async function to throw a sync error, can you merge with master and and push changes again?

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js [hunk 4]
```diff
@@ -1334,3 +1336,121 @@ describe('AwsCompileEventBridgeEvents', () => {
...
+        )
+        .then(({ cfTemplate, awsNaming }) => {
+          ({ Resources: cfResources } = cfTemplate);
+          naming = awsNaming;
+        })
+    );
+
+    it('should create the correct execution role', () => {
+      const executionRolePolicies =
+        cfResources.IamRoleCustomResourcesLambdaExecution.Properties.Policies;
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/eventBridge/index.test.js:1377]: We should use `naming` to resolve resource logical ids (then changes in them won't make test fail)

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js [hunk 5]
```diff
@@ -1334,3 +1336,121 @@ describe('AwsCompileEventBridgeEvents', () => {
...
+                {
+                  Ref: 'AWS::AccountId',
+                },
+                'function',
+                'service-dev-foo',
+              ],
+            ],
+          },
+        },
+      ]);
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/eventBridge/index.test.js:1427]: Let's maybe not confirm on whole structure (it's problematic maintenance wise) but instead simply test that there's a policy which allows `'events:PutRule'` on rule of which ARN ends with `rule/service-dev-foo-rule-1`

File: lib/plugins/aws/package/compile/events/eventBridge/index.test.js [hunk 6]
```diff
@@ -1334,3 +1336,121 @@ describe('AwsCompileEventBridgeEvents', () => {
...
+            EventBus: 'arn:aws:events:us-east-1:12345:event-bus/default',
+            Input: undefined,
+            InputPath: undefined,
+            InputTransformer: undefined,
+            RuleName: 'service-dev-foo-rule-1',
+            Pattern: undefined,
+            Schedule: 'rate(10 minutes)',
+          },
+        },
+      });
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/eventBridge/index.test.js:1453]: Same here we should use `naming` for logical id resolution, and confirm just on few core properties

Comment (TheEdgeOfCat) [reply lib/plugins/aws/package/compile/events/eventBridge/index.test.js]: I have made some changes, but I wasn't sure which names I need to extract from the naming object. Let me know if I should add any of the other ones.

**Comment by TheEdgeOfCat** [comment]
I have ran the test twice now. The first time only windows failed. Now node 8 too. I have no idea why and how, there is no log to indicate anything.

**Comment by TheEdgeOfCat** [comment]
> I will further investigate CI fail, it's very weird (looks as GitHub actions issue)

When I run the test locally it takes quite a while. 10-15 seconds for the test that use runServerless. Maybe there is a timeout for the test and in combination with GitHub giving you less CPU than what I have locally it takes longer than the timeout.

Although I'm still not sure why it takes that long. All the other runServerless tests run just fine, only the eventbridge one takes much longer. I'm not sure whether it's because of my test, or something else.

**Comment by TheEdgeOfCat** [comment]
I would never have guessed that that's the problem. Thank you for the fix, I applied it and now the tests run a lot faster and the GH actions also seem to work now.

**Pull request merged**